packages/node_modules/pouchdb-find/src/adapters/http/index.js

```diff
@@ -1,35 +1,50 @@
+import { generateErrorFromResponse } from 'pouchdb-errors';
+import { Headers } from 'pouchdb-fetch';
 import massageCreateIndexRequest from '../../massageCreateIndexRequest';
 
+function dbFetch(db, path, opts, callback) {
+  var status, ok;
+  opts.headers = new Headers({'Content-type': 'application/json'});
+  db.fetch(path, opts).then(function (response) {
+    status = response.status;
+    ok = response.ok;
+    return response.json();
+  }).then(function (json) {
+    if (!ok) {
+      json.status = status;
+      var err = generateErrorFromResponse(json);
+      callback(err);
+    } else {
+      callback(null, json);
+    }
+  });
```

Comment:
Same as before, if there is a network error or a json parsing error the callback will not be called.

Reply:
This one however the .catch is needed I think, thanks